[Fong Yi Fei] iP

YiFei2 opened the pull request:
# It's Bobby.
### Features
- Bobby will remember what you tell him.
- Bobby can remind you of what you told him.
- Bobby can search his memory for something you told him.
### What can Bobby remember?
1. To dos
2. Deadlines (with date in YYYY-MM-DD format)
3. Events (with date/time in any format)
### Here is how Bobby would encourage you in `taskDone`
```
public static void taskDone(Task t) {
        System.out.println("Bobby applauds you. This task is done:\n" + t);
    }
```
### An example of what you could do with Bobby.
- [x] CS2103 iP tab
- [x] CS2106 lab
- [ ] **_shower_** 🚿 
- [x] CS2102 tutorial

> "If you got a dog, feed it everyday. But if you don't, don't."

[?](https://www.goodhousekeeping.com/life/pets/g4531/cutest-dog-breeds/)

Commit:
Add Gradle support

YiFei2 committed:
changed name of chatbot, completed level 1

YiFei2 committed:
completed level-2

YiFei2 committed:
fixed level-2, included print when item is added to list

YiFei2 committed:
Implemented task.java, completed Level-3

YiFei2 committed:
added Deadline, Event, Todo classes. Updated toString method in Task and cleaned up print statements in Bobby.

YiFei2 committed:
completed Level-4

YiFei2 committed:
added test cases, fixed exception occurring due to scanner object having no next line

YiFei2 committed:
Created methods for adding Task to list and methods now throw BobbyExceptions and are handled in the main method

YiFei2 committed:
implemented delete method, completed level-6

YiFei2 committed:
add use of enum, create new bobby.txt file at start of program

YiFei2 committed:
add writeToFile

YiFei2 committed:
add writeToFile() methods to various tasks, implement write and loading for bobby

YiFei2 committed:
add date with LocalDate type for deadlines

YiFei2 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java

YiFei2 committed:
extract out Parser, Ui, Tasklist, and Storage classes to apply OOP principles

YiFei2 committed:
create bobby package

YiFei2 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

YiFei2 committed:
add junit tests for todo, dealine and invalid input

YiFei2 committed:
change build name to bobby

YiFei2 committed:
clean up some code, insert javadocs for bobby.java

YiFei2 committed:
write javadocs for Parser, Bobby, Storage and TaskList classes

YiFei2 committed:
tweak code to adhere to CodingStandards, add and run checkstyle

YiFei2 committed:
add Find function

YiFei2 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/bobby/Bobby.java

YiFei2 committed:
resolve merge conflicts

YiFei2 committed:
Merge branch 'branch-level-9'

YiFei2 committed:
insert javadocs for Find

khoahre123 commented on src/main/java/bobby/Event.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Override
+    public void writeToFile(FileWriter fw) throws IOException {
+        fw.write("E ; " + isDone + " ; " + description + " ; " + at + System.lineSeparator());
```
Should the break line be 8 spaces indent compared to the parent line?

khoahre123 commented on src/main/java/bobby/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
Maybe another more meaningful name?

khoahre123 commented on src/main/java/bobby/Parser.java:
```diff
@@ -0,0 +1,99 @@
+            case DEADLINE:
+                try {
+                    tasks.addDeadline(userInput);
+                } catch (BobbyException e) {
```
I think you can combine all the expected exceptions into one file and call that instance

khoahre123 commented on src/main/java/bobby/TaskList.java:
```diff
@@ -0,0 +1,173 @@
+                String description = splitInputs[0];
+                LocalDate by = LocalDate.parse(splitInputs[1]);
+                Deadline newDeadline = new Deadline(description, by);
+                System.out.println("Bobby heard: " + newDeadline);
```
Maybe you can take all the ```System.out.println``` line into a new method in Ui class?

khoahre123 commented on src/main/java/bobby/BobbyException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class BobbyException extends Exception {
+    public BobbyException(String errorMessage) {
+        super(errorMessage);
```
I think you can turn this into a more customize file, instead of simply extends the Exception class

khoahre123 commented on src/main/java/bobby/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
Should the break line be 8 spaces indent compared to the parent line?

khoahre123 commented on src/main/java/bobby/Parser.java:
```diff
@@ -0,0 +1,99 @@
+                } catch (BobbyException e) {
+                    System.out.println(e);
+                } catch (DateTimeParseException e) {
+                    System.out.println("Invalid date format. Please use YYYY-MM-DD.");
```
I noticed the same issue in several other places too

khoahre123 reviewed (commented):
Look pretty solid, some lines need to change according to the coding standard. Overall, good job 💯

YiFei2 committed:
add MainWindow, Main, Launcher, DialogBox classes for javafx GUI implementation

YiFei2 committed:
Change Parse in Parser to return String type, implemented getResponse() method in Bobby

YiFei2 committed:
Refine GUI to display welcome message and allow user to exit using "bye" command
Change GUI display pictures
Ensure text is displayed fully for long tasklists

YiFei2 committed:
fix minor bugs with dialogbox and building jar file

YiFei2 committed:
enable assertions

YiFei2 committed:
Current code do not have assertions

To ensure assumptions and ease of debugging for future extensions

Let's insert assert statements in various points to ensure
assumptions hold and methods function as expected

YiFei2 committed:
Merge pull request #1 from YiFei2/branch-A-Assertions

Branch a assertions

YiFei2 committed:
improve code quality

YiFei2 committed:
Merge branch 'master' of https://github.com/YiFei2/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/bobby/Parser.java

YiFei2 committed:
resolve merge conflicts

YiFei2 committed:
Revert "resolve merge conflicts"

This reverts commit 5b234ed55ef54a334665eb195ac301d8dbd768d3.

YiFei2 committed:
Parser class: extract out methods to do splitting

Parse() method in Parser class do not have same level of abstraction
and have common behaviors across several switch cases.

Extracting out common behaviors into private methods in the class
improves the code quality of the method as it now has single level
of abstraction and do not have common behavior across cases.

Let's extract out the splitting process into various private methods
to split the input

YiFei2 committed:
extract creation of file in loadFile() method to resolve arrowhead
style code

YiFei2 committed:
remove unused method

YiFei2 committed:
Merge pull request #2 from YiFei2/branch-A-CodeQuality

Branch a code quality

YiFei2 committed:
Clean up code, implement State class and stateStack

YiFei2 committed:
implement stateStack

YiFei2 committed:
implement undo

YiFei2 committed:
Implement Undo command

Undo command will help users correct their incorrect inputs easily.

Let's keep track of the state of Bobby after each change made and
implement the Undo command to reverse the most recent change.

YiFei2 committed:
Fix up javadocs and coding standard issues

YiFei2 committed:
Edit Readme.md and add Ui.png

YiFei2 committed:
Set theme jekyll-theme-slate

YiFei2 committed:
Update README.md

YiFei2 committed:
Update README.md

YiFei2 committed:
Update README.md

YiFei2 committed:
Update README.md

YiFei2 committed:
rename Ui.png to Ui

YiFei2 committed:
Merge branch 'master' of https://github.com/YiFei2/ip
